### PR TITLE
chore: Migrate to supabase v2

### LINKED ISSUE #20: [ISAAC-486] Upgrade to supabase-js v2
<sub>From [SyncLinear.com](https://synclinear.com) | [ISAAC-486](https://linear.app/isaac-editor/issue/ISAAC-486/upgrade-to-supabase-js-v2)</sub>

Update all instances of JS Supabase sdk in our application from the currently used supabase-js v1 to the supabase-js v2. This upgrade will involve modifying existing Supabase integrations to ensure compatibility with the latest version, addressing any breaking changes, and potentially leveraging new features and improvements offered in v2.

### PR DESCRIPTION
**Pull request by ShaneMaglangit**
> Note: Line diff is humongous because of `package-json.lock` with `+-27,308` (lint + package updates)

## Description

This PR migrates our Supabase integration from v1 to v2. The update includes changes to the SDK or API to accommodate the new features and improvements introduced in Supabase v2.

## Suggestions for improvements

- Add automated tests to cover for, at the very least, the primary workflows -- which should allow us to have confidence in doing these massive steps in the future without having to be anxious of breakages that might get introduced.
  - I can't guarantee that this PR does not introduce any bugs or errors due to the nature of massive migration (practically impossible to do this incrementally) and lack of test coverage.
  - I can only guarantee that I've performed the due diligence such as manual testing and static code analysis to ensure most features still works as intended.
  - E2E tests with frameworks like Playwright is plenty if doing lower level tests is too cumbersome given this project's current scale.

Fixes #20 
/claim #20

### CODE REVIEW

File: web/hooks/api/useInitializeUserUploads.tsx [hunk 1]
```diff
@@ -73,23 +73,32 @@ const useInitializeUserUploads = (targetProjectId?: string) => {
...
+					table: 'uploads',
+					where: { user_id: userId },
+				},
+				handleRealtimeUpdates,
+			)
 			.subscribe();
 
 		return () => {
-			supabase.removeSubscription(subscription);
+			void supabase.removeChannel(subscription);
```
Comment (ShaneMaglangit) [inline web/hooks/api/useInitializeUserUploads.tsx:101]: Not sure if this is the correct v2 version of this but won't bother fixing:
- Deprecated function
- No existing usage (safe to remove, unless we have a specific reason for keeping it)

**Review by eimenhmdt — CHANGES_REQUESTED** [review]
Thanks for the PR!

The code looks good in general and everything seems to function as intended. However, I came across a small bug: When clicking sign out, the landing page only updates after refreshing and then renders the correct state.

CleanShot 2024-01-27 at 19 23 24

**Comment by eimenhmdt** [comment]
Thank you for the contribution. I reviewed the code. Also, I fully agree with the need to implement proper testing.

**Comment by ShaneMaglangit** [comment]
> Thanks for the PR!
> 
> The code looks good in general and everything seems to function as intended. However, I came across a small bug: When clicking sign out, the landing page only updates after refreshing and then renders the correct state.
> 
> CleanShot 2024-01-27 at 19 23 24 [ CleanShot 2024-01-27 at 19 23 24 ](https://private-user-images.githubusercontent.com/64800243/300199743-3245ab84-c403-4fa7-ab3b-9f99cc11f14e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY0MDc5MTgsIm5iZiI6MTcwNjQwNzYxOCwicGF0aCI6Ii82NDgwMDI0My8zMDAxOTk3NDMtMzI0NWFiODQtYzQwMy00ZmE3LWFiM2ItOWY5OWNjMTFmMTRlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI4VDAyMDY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4MDBiYmQ1YWVhYTRhMjQwYjczOWRjYjRjYmI3MzlkMWQzZTYxOGU0YjEyZWUzMTlkM2Y5ZDg3MTEyM2IyOTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3FMH7HfBjaFYLfl1I3-wnnIGzLFAlAqDBEkXtvzhIVw) [ ](https://private-user-images.githubusercontent.com/64800243/300199743-3245ab84-c403-4fa7-ab3b-9f99cc11f14e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY0MDc5MTgsIm5iZiI6MTcwNjQwNzYxOCwicGF0aCI6Ii82NDgwMDI0My8zMDAxOTk3NDMtMzI0NWFiODQtYzQwMy00ZmE3LWFiM2ItOWY5OWNjMTFmMTRlLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI4VDAyMDY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4MDBiYmQ1YWVhYTRhMjQwYjczOWRjYjRjYmI3MzlkMWQzZTYxOGU0YjEyZWUzMTlkM2Y5ZDg3MTEyM2IyOTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3FMH7HfBjaFYLfl1I3-wnnIGzLFAlAqDBEkXtvzhIVw)

Great catch! I've pushed a commit to fix this. Will appreciate if you can have another look @eimenhmdt :boomerang:

**Review by eimenhmdt — APPROVED** [review]
lgtm now

**Comment by eimenhmdt** [comment]
@ShaneMaglangit I'm getting a build error when trying to deploy the merged code:

`
Type error: Property 'weak_password' does not exist on type '{ user: User; session: Session; weak_password?: WeakPassword; } \| { user: null; session: null; }'.
--
08:02:16.170 | Property 'weak_password' does not exist on type '{ user: null; session: null; }'.
08:02:16.170 |  
08:02:16.170 | 189 \|     data.weak_password.reasons.reduce((a: boolean, i: any) => a && typeof i === 'string', true)
08:02:16.170 | 190 \|   ) {
08:02:16.170 | > 191 \|     response.data.weak_password = data.weak_password
08:02:16.170 | \|                   ^
08:02:16.170 | 192 \|   }
08:02:16.170 | 193 \|
08:02:16.170 | 194 \|   return response
08:02:16.297 | Error: Command "npm run build" exited with 1
`

Do you have any idea what causes this?

**Comment by ShaneMaglangit** [comment]
> I'm getting a build error when trying to deploy the merged code:
@eimenhmdt hmm let me check

**Comment by ShaneMaglangit** [comment]
@eimenhmdt what file/line is this? can't find anything with grep.

**Pull request merged**